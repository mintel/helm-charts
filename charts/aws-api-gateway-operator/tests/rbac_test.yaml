suite: test rbac
templates:
  - role.yaml
  - role-binding.yaml
  - cluster-role.yaml
  - cluster-role-binding.yaml
tests:
  - it: role can access awsapigateway objects
    template: cluster-role.yaml
    set:
      image.tag: latest
    asserts:
      - isKind:
          of: ClusterRole
      - equal:
          path: rules[0].resources
          value:
            - awsapigateways
      - equal:
          path: rules[0].verbs
          value: [create, delete, get, list, patch, update, watch]
