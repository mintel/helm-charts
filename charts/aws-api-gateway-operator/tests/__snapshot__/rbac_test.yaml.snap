role can access awsapigateway objects:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      annotations:
        helm.sh/chart: aws-api-gateway-operator-0.1.2
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: aws-api-gateway-operator
        app.kubernetes.io/part-of: aws-api-gateway-operator
        app.mintel.com/env: ${CLUSTER_ENV}
        app.mintel.com/owner: SRE
        app.mintel.com/region: ${CLUSTER_REGION}
        name: aws-api-gateway-operator
      name: aws-api-gateway-operator-role
    rules:
      - apiGroups:
          - mintel.cloud
        resources:
          - awsapigateways
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - events
        verbs:
          - create
          - patch
      - apiGroups:
          - ""
        resources:
          - pods
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - ""
        resources:
          - nodes
          - namespaces
          - endpoints
        verbs:
          - get
          - list
          - watch
